### ISSUE TITLE
Speed between spark.randomforset and mml.lightgbm

### ISSUE THREAD

**Issue by RainFung** [Azure/mmlspark#608]
Speed between spark.randomforset and mml.lightgbm，which is faster?

**Comment by imatiach-msft** [comment]
@RainFung the doc states:
https://github.com/Azure/mmlspark/blob/master/docs/lightgbm.md
```
LightGBM on Spark is 10-30% faster than SparkML on the Higgs dataset, and achieves a 15% increase in AUC
```
It varies a lot from cluster to cluster and based on parameters though, and I haven't done any performance comparisons recently.
In general, I think a better comparison would be between lightgbm and xgboost (which also has a spark implementation).  I would try all 3 and see which one is best on your dataset, as performance could vary greatly on different datasets.

I've put the azure databricks and HDInsight notebooks here for the Higgs dataset in my branch (note I will not be checking these in):
https://github.com/imatiach-msft/mmlspark/tree/ilmat/higgs-notebooks/notebooks/higgs
You can get the Higgs csv dataset here:
https://github.com/guolinke/boosting_tree_benchmarks/tree/master/data
Specifically this link:
https://archive.ics.uci.edu/ml/datasets/HIGGS

**Comment by tyler-romero** [comment]
Also note that Random Forests and Gradient Boosted Trees/Machines are different classes of algorithms. Another better comparison would be between mml.LightGBM and spark.GradientBoostedTrees.

Note that Gradient Boosted Trees are more expressive than Random Forests.

From [the databricks blog](https://databricks.com/blog/2015/01/21/random-forests-and-boosting-in-mllib.html):

> Random Forests: Since each tree in a Random Forest is trained independently, multiple trees can be trained in parallel (in addition to the parallelization for single trees). MLlib does exactly that: A variable number of sub-trees are trained in parallel, where the number is optimized on each iteration based on memory constraints.
>
> GBTs: Since GBTs must train one tree at a time, training is only parallelized at the single tree level.

and

> Comparing the two methods, Random Forests are faster to train, but they often require deeper trees than GBTs to achieve the same error. GBTs can further reduce the error with each iteration, but they can begin to overfit (increase test error) after too many iterations. Random Forests do not overfit as easily, but their test error plateaus.

**Comment by RainFung** [comment]
Thanks I'll try.
